lib/matplotlib/scale.py

```diff
@@ -82,6 +82,88 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class ArbitraryTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: The forward function for the transform
+
+         inverse: The inverse of the forward function.
+         """
+         super().__init__()
+         if callable(forward) and callable(inverse):
+             self._forward = forward
+             self._inverse = inverse
+         else:
+             raise ValueError('arguments to ArbitraryTransform must '
+                              'be functions.')
+
+     def transform_non_affine(self, values):
+         with np.errstate(divide='ignore', invalid='ignore'):
```

Comment:
I think setting the errstate should be the job of the function itself?  (aka. let's assume that whoever passes arbitrary functions in knows what they're doing, or document that they should)

Reply:
Sounds good to me.  That was leftover from LogScale (I think), and I didn't quite grok what it was doing...  I'll need to make an example that does the error checking though....

Reply:
well, in logscale it effectively belongs to the user-defined (meaning by us) function